### DISCUSSION TITLE
[v5] referencing another field in unique validation rule

### DISCUSSION THREAD

**Discussion by popovicieduard** [adonisjs/core#2487] [Help] [answered]
I'm trying to reference another field in a schema validator class but can't find a solution to make it work, is this possible?

```js
    accountId: schema.number(),
    emailAddress: schema.string({}, [
      rules.email(),
      rules.minLength(3),
      rules.maxLength(80),
      rules.unique({
        table: 'contacts',
        column: 'email_address',
        where: { account_id: 'accountId' }, // How can I reference the accountId field from above ?
      }),
    ]),
```

**Comment by dqgiang** [comment]
Have you tried `refs` to check for value constraint at runtime?
https://preview.adonisjs.com/guides/validator/rules#additional-constraints
https://preview.adonisjs.com/guides/validator/schema-caching#option-3-using-refs

The docs pretty much cover usage inside HTTP request cycle.
I haven't tried with standalone usage. I guess it would involve passing the data into local properties.

**Comment by arthurcvm** [comment, marked as answer]
Hi @popovicieduard you solve this? I have same problem, I need pass to custom rule the value of another field in validator.

**Reply by popovicieduard** [reply]
Managed to do this with a custom rule.

**Reply by arthurcvm** [reply]
Hi! I solved this passing the data in object and validate the object in custom rule, this is the simplest solution I found, look the code:

In validator I send this object:
```typescript
related: schema.object([rules.parentHasPermission()]).members({
    parent_permission_group_id: schema.number.optional([
      rules.exists({ table: 't_permission_groups', column: 'id' })
    ]),
    permissions: schema.array.optional().members(schema.number())
  })
```

In custom rule I check if parent_permission_group has relationship with  sent permissions.
